### DISCUSSION TITLE
Cmake & OpenSSL

### DISCUSSION THREAD

**Discussion by sorydima** [codemagic-ci-cd/codemagic-docs#2527] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Run & Compile REChain-.git for Windows 

### Expected results

build done..

### Actual results

CMake Error at C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY) (found
  version "3.2.1")
Call Stack (most recent call first):
  C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.28/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.28/Modules/FindOpenSSL.cmake:668 (find_package_handle_standard_args)
  flutter/ephemeral/.plugin_symlinks/sqlcipher_flutter_libs/windows/CMakeLists.txt:27 (find_package)

**Comment by sorydima** [comment]
So on, here the error. 
![Снимок экрана 2024-05-15 142259](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/7fbeceb1-c2e3-4723-a475-b662b0c1fc36)

**Comment by lpusz** [comment]
We encountered exactly the same problem in our Flutter project. So far we have not been able to find any solution or workaround to the issue.

**Reply by sorydima** [reply]
Any tries to reach out the goal?)) So it's flutter 3.19.6 , Win 11 Latest, OpenSSL 3.3.0 Full

**Reply by StolarczykJakub** [reply]
flutter: 3.16.5
  Setup which is available out-of-the-box on Codemagic Windows build machines.
  OpenSSL 3.2.1 30 Jan 2024 (Library: OpenSSL 3.2.1 30 Jan 2024)
  
  We tried to update, reinstall, downgrade, set path, add to env variables. 
  But nothing worked.

**Reply by sorydima** [reply]
Manually compile crypto library? Not default OpenSSL but crypto part of it from GitHub repo? Cause I don't understand what really happening 😭

**Reply by sorydima** [reply]
>   flutter: 3.16.5
>   Setup which is available out-of-the-box on Codemagic Windows build machines.
>   OpenSSL 3.2.1 30 Jan 2024 (Library: OpenSSL 3.2.1 30 Jan 2024)
>   
>   We tried to update, reinstall, downgrade, set path, add to env variables. 
>   But nothing worked.

Maybe Win 10 will fix it. So it's the problem with Windows update? 🤔

**Reply by sorydima** [reply]
> We encountered exactly the same problem in our Flutter project. So far we have not been able to find any solution or workaround to the issue.

Maybe Win 10 will fix it. So it's the problem with Windows update? 🤔

**Reply by Vadym-Barda** [reply]
It looks like there is some global issue with SSL. I see similar issue with building Android and iOS application

**Comment by helinanever** [comment]
Thanks everyone for raising this, we're looking into it.

**Reply by sorydima** [reply]
Thank you 🙏😊 Waiting for the release ☺️

**Reply by StolarczykJakub** [reply]
Thanks for quick response.

**Comment by helinanever** [comment]
Would any of you share your build IDs so we can investigate further? We haven't been able to reproduce the issue so far.

**Reply by StolarczykJakub** [reply]
Here you go: 6645cafaaaee750644346401
Step: Build windows release.
Line: 858

**Reply by sorydima** [reply]
> Would any of you share your build IDs so we can investigate further? We haven't been able to reproduce the issue so far. 

Any news on the error?) 😕

**Comment by mohammedbabelly20** [comment, marked as answer]
Hello everyone, sorry for the inconvenience but it seems there's an [issue](http://disq.us/p/2xvyml9) with the **OpenSSL 3.2.1** so, we had to downgrade to the same version we used to have before the latest [Windows image upgrade](https://github.com/orgs/codemagic-ci-cd/discussions/2474).
The issue should be resolved now.

**Reply by sorydima** [reply]
So now it will work in CodeMagic Windows Machine?

**Reply by AndriiChemer** [reply]
@mohammedbabelly20 thanks, it works

**Reply by sorydima** [reply]
Great 😃

**Reply by StolarczykJakub** [reply]
@mohammedbabelly20 I have an idea. How about **including build machine specification into build overview**? (History of Pre-installed tools) It could be in some "shink expand" toggle. We would be able to check what is the setup for current build that's failing, and what was the setup for last build that worked. It would be helpful and speed up the bug finding process.

**Reply by sorydima** [reply]
> @mohammedbabelly20 I have an idea. How about **including build machine specification into build overview**? (History of Pre-installed tools) It could be in some "shink expand" toggle. We would be able to check what is the setup for current build that's failing, and what was the setup for last build that worked. It would be helpful and speed up the bug finding process. 

Yeah 👍, Sounds good 😊

**Comment by sorydima** [comment]
Thank you guys! Now REChain- build is working pretty fine! ID 6647536bbedc73a166b22416, 
https://github.com/sorydima/REChain-/commit/b892326bfa0ed4e8f4a138a82a078da7a1ef7dca
Our proj - https://github.com/sorydima/REChain- !
So also if anyone want to collaborate with us and distribute apps through stores such as F-Droid, Snacraft, Flathub, etc.. - write me to @sorydima:matrix.org , tg - @sorydima, ig - @sorydima 
Also here is our Win10H2 Build! 

![Снимок экрана от 2024-05-17 16-02-53](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/905df23a-5bcd-46d8-bcf2-59b4cdbe86a7)

https://api.codemagic.io/artifacts/2387ae9e-1ccf-4b4e-8093-6b35cdd51e0d/b4d1a051-5f8b-4c75-b8a8-2efa385e2997/release.zip

**Reply by sorydima** [reply]
Guys, so I have a problem again 😭😞. It's okay with compiling and building the installer with InnoSetup. It's okay , but on Win 10 latest with Oracle VM it doesn't work. So check https://github.com/sorydima/REChain-/blob/main/windows/rechainonline.exe 
It's the installer that has been done with InnoSetup & https://github.com/sorydima/REChain-/tree/main/windows/release it's the release build from CodeMagic! So any thoughts 💭💬?)
